### PR TITLE
[modules] Approach moving target fixes

### PR DESCRIPTION
**Pull request by dewagter**
- allow storing specific gains in airframe files
 - put all settings in the struct 
 - remove unused

### CODE REVIEW

File: conf/modules/approach_moving_target.xml [hunk 1]
```diff
@@ -5,21 +5,29 @@
     <description>
       Approach a moving target (e.g. ship) along a diagonal.
     </description>
-    <define name="AMT_ERR_SLOWDOWN_GAIN" value="0.25" description="slowing down when arriving"/>
     <define name="CUTOFF_FREQ_FILTERS_HZ" value="0.5" description="low pass filter on WP position"/>
-  </doc>
+
+    <define name="APPROACH_MOVING_TARGET_SLOPE" value="35.0" description="[degrees] the angle of the slope for the approach"/>
```
Comment (fvantienen) [inline conf/modules/approach_moving_target.xml:10]: Maybe add a section?

File: sw/airborne/modules/ctrl/approach_moving_target.c [hunk 2]
```diff
@@ -54,13 +75,15 @@ Butterworth2LowPass target_heading_filt;
...
   .psi_ref = 180.0,
-  .slope_ref = 35.0,
-  .speed_gain = 1.0,
-  .relvel_gain = 1.0,
+  .slope_ref = APPROACH_MOVING_TARGET_SLOPE,
+  .err_slowdown_gain = APPROACH_MOVING_TARGET_ERR_SLOWDOWN_GAIN,
+  .pos_gain = APPROACH_MOVING_TARGET_POS_GAIN,
+  .speed_gain = APPROACH_MOVING_TARGET_SPEED_GAIN,
+  .relvel_gain = APPROACH_MOVING_TARGET_RELVEL_GAIN,
+  .cutoff_freq_filters_hz = CUTOFF_FREQ_FILTERS_HZ,
```
Comment (fvantienen) [inline sw/airborne/modules/ctrl/approach_moving_target.c:86]: Doesn't have the prefix `APPROACH_MOVING_TARGET` and also no documentation etc.

**Review by fvantienen — CHANGES_REQUESTED** [review]

File: conf/modules/approach_moving_target.xml [hunk 3]
```diff
@@ -5,21 +5,29 @@
...
+    <define name="APPROACH_MOVING_TARGET_DISTANCE" value="60.0" description="[m] the initial distance to start the approach"/>
+    <define name="APPROACH_MOVING_TARGET_SPEED" value="-1.0" description="[m/s] the approach speed along the slope"/>
+
+    <define name="APPROACH_MOVING_TARGET_ERR_SLOWDOWN_GAIN" value="0.25" description="slowing down when arriving"/>
```
Comment (fvantienen) [inline conf/modules/approach_moving_target.xml:14]: Maybe also say which direction does what

File: conf/modules/approach_moving_target.xml (hunk 1 above)
Comment (dewagter) [reply conf/modules/approach_moving_target.xml]: done

File: sw/airborne/modules/ctrl/approach_moving_target.c (hunk 2 above)
Comment (dewagter) [reply sw/airborne/modules/ctrl/approach_moving_target.c]: done

**Comment by dewagter** [comment]
This one should be ready. Tested on nederdrones.

**Review by fvantienen — APPROVED** [review]

**Pull request merged**